org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectRegistryReader.java

```diff
@@ -65,53 +63,54 @@ public ProjectRegistryReader(File stateLocationDir) {
   public ProjectRegistry readWorkspaceState(final ProjectRegistryManager managerImpl) {
     if(stateFile.exists()) {
       final PackageAdmin packageAdmin = getPackageAdmin();
-      ObjectInputStream is = null;
-      try {
-        is = new ObjectInputStream(new BufferedInputStream(new FileInputStream(stateFile))) {
```

Comment:
The creation of the ObjectInputStream was simply extracted into an own class to improve readability.
The same was done for the ObjectOutputStream below.
I think an anonymous class within a try-with-resources would not be very readable.

Reply:
:+1: